src/test/java/htsjdk/utils/TestNGUtilsTest.java

```diff
@@ -0,0 +1,95 @@
+package htsjdk.utils;
+
+import htsjdk.HtsjdkTest;
+import org.testng.Assert;
+import org.testng.annotations.DataProvider;
+import org.testng.annotations.Test;
+
+import java.util.Arrays;
+import java.util.List;
+
+public class TestNGUtilsTest extends HtsjdkTest {
+
+    @DataProvider
+    public Object[][] getArraysAndLists() {
+        return new Object[][]{
+                {
+                        new Object[][]{
+                                {1, 2},
+                                {3, 4}
+                        }
+                        , Arrays.asList(
+                        Arrays.asList(1, 2),
+                        Arrays.asList(3, 4)
+                )
+                },
+                {
+                        new Object[][]{
+                                {1}
+                        },
+                        Arrays.asList(
+                                Arrays.asList(1)
+                        )
+                },
+                {
+                        new Object[][]{
+                                {1,2,3},
+                                {4,5,6}
+                        },
+                        Arrays.asList(
+                                Arrays.asList(1,2,3),
+                                Arrays.asList(4,5,6)
+                        )
+                },
+                {
+                        new Object[][]{
+                                {1},
+                                {2},
+                                {3},
+                                {4}
+                        },
+                        Arrays.asList(
+                                Arrays.asList(1),
+                                Arrays.asList(2),
+                                Arrays.asList(3),
+                                Arrays.asList(4)
+                        )
+                }
+        };
+    }
+
+    @Test(dataProvider = "getArraysAndLists")
+    public void testObjectsToLists(Object[][] objects, List<List<Object>> lists){
```

Comment:
Many cases here where the code is missing a space before a `{`, and missing a space after a comma above (e.g. line 36). It might be easier to reformat the file completely in the IDE.

Reply:
done... I'm not sure it helps much though, this is a gross block of cases